Assets/Scripts/Algorithms/Prediction/LinearExtrapolator.cs

```diff
@@ -8,6 +8,10 @@ public LinearExtrapolator(IHierarchical hierarchical) : base(hierarchical) {}
   // Predict the future state of the hierarchical object by linearly extrapolating its current
   // position and velocity.
   public override PredictorState Predict(float time) {
+    if (Hierarchical == null) {
+      return new PredictorState();
+    }
+
```

Comment:
Returning a default/empty `PredictorState` seems misleading since callers cannot distinguish between "prediction failed" and "predicted position is (0,0,0)"

Recommend throwing an exception or returning null and updating the callers to handle null. At the least log a warning/error.